sympy/geometry/line.py

```diff
@@ -1083,24 +1093,67 @@ class Line(LinearEntity):
     >>> s = Segment((0, 0), (0, 1))
     >>> Line(s).equation()
     x
+
+    >>> # a circle object is returned
+    >>> from sympy.abc import x, y, a, b
+    >>> Line(3 * x + y + 18)
+    Line2D(Point2D(0, 0), Point2D(1, -3))
+    >>> Line(3 * a + b + 18, x='a', y='b')
+    Line2D(Point2D(0, 0), Point2D(1, -3))
     """
 
-    def __new__(cls, p1, p2=None, **kwargs):
-        if isinstance(p1, LinearEntity):
-            if p2:
-                raise ValueError('If p1 is a LinearEntity, p2 must be None.')
-            dim = len(p1.p1)
+    def __new__(cls, *args, **kwargs):
+
+        if isinstance(args[0], tuple) or isinstance(args[0], list) or isinstance(args[0], Point) or isinstance(args[0], LinearEntity):
```

Comment:
In both cases I think you can just sympify the args and then detect equation with 
```python
if len(args) == 1 and isinstance(args[0], Expr):
    # equation case
else:
    # normal case
```

Reply:
@smichr Yes you are right. I should've done that earlier. Thanks for the input. Please take a look now.